crates/store/re_sorbet/src/chunk_batch.rs

```diff
@@ -51,58 +219,109 @@ impl From<ChunkBatch> for ArrowRecordBatch {
-        Ok(Self { entity_path, batch })
+        let batch = ArrowRecordBatch::try_new_with_options(
+            arrow_schema.into(),
+            batch.columns().to_vec(),
+            &RecordBatchOptions::default().with_row_count(Some(batch.num_rows())),
+        )
+        .expect("Can't fail");
+
+        Ok(Self {
+            schema: chunk_schema,
+            batch,
+        })
     }
 }
 
-/// ## Metadata keys for the record batch metadata
-impl ChunkBatch {
-    /// The key used to identify the version of the Rerun schema.
-    const CHUNK_METADATA_KEY_VERSION: &'static str = "rerun.version";
+/// Make sure all data columns are `ListArrays`.
+fn make_all_data_columns_list_arrays(batch: &ArrowRecordBatch) -> ArrowRecordBatch {
+    re_tracing::profile_function!();
 
-    /// The version of the Rerun schema.
-    const CHUNK_METADATA_VERSION: &'static str = "1";
+    let num_columns = batch.num_columns();
+    let mut fields: Vec<ArrowFieldRef> = Vec::with_capacity(num_columns);
+    let mut columns: Vec<ArrowArrayRef> = Vec::with_capacity(num_columns);
 
-    /// The key used to identify a Rerun [`EntityPath`] in the record batch metadata.
-    const CHUNK_METADATA_KEY_ENTITY_PATH: &'static str = "rerun.entity_path";
-}
+    for (field, data) in itertools::izip!(batch.schema().fields(), batch.columns()) {
+        if data
+            .downcast_array_ref::<arrow::array::ListArray>()
+            .is_some()
+        {
+            // Already fine
+            fields.push(field.clone());
+            columns.push(data.clone());
+        } else if field
+            .metadata()
+            .get("rerun.kind")
+            .is_some_and(|kind| kind == "data")
+        {
+            // We slice each column array into individual arrays and then convert the whole lot into a ListArray
 
-impl ChunkBatch {
-    /// Returns the [`EntityPath`] of the batch.
-    #[inline]
-    pub fn entity_path(&self) -> &EntityPath {
-        &self.entity_path
+            let data_field_inner =
+                ArrowField::new("item", field.data_type().clone(), true /* nullable */);
+
+            let data_field = ArrowField::new(
+                field.name().clone(),
+                ArrowDataType::List(Arc::new(data_field_inner.clone())),
+                false, /* not nullable */
+            )
+            .with_metadata(field.metadata().clone());
+
+            let mut sliced: Vec<ArrowArrayRef> = Vec::new();
+            for idx in 0..data.len() {
+                sliced.push(data.clone().slice(idx, 1));
+            }
+
+            let data_arrays = sliced.iter().map(|e| Some(e.as_ref())).collect::<Vec<_>>();
+            #[allow(clippy::unwrap_used)] // we know we've given the right field type
+            let list_array: arrow::array::ListArray =
+                re_arrow_util::arrow_util::arrays_to_list_array(
+                    data_field_inner.data_type().clone(),
+                    &data_arrays,
+                )
+                .unwrap();
```

Comment:
I'm surprised this looks this costly/complicated? This should just be a matter of creating a listarray with unit offsets no?
E.g. in pseudo-rust:
```rust
let offsets = std::iter::repeat(1).take(data.len());
let list_array = ListArray::new(data_field, offsets, data, None);
```

Reply:
I'll look at it in separate PR (this one is big enough already). But I'll add a TODO.